dependency-convergence/src/main/java/DependencyUpdateTest.java

```diff
@@ -0,0 +1,344 @@
+  private static final List<ArtifactData> unfindableClientLibraries = new ArrayList<>();
+
+  private static final List[] librariesClassified = new List[]{
+      successfulClientLibraries, librariesWithoutSharedDeps,
+      librariesWithBadSharedDepsVersion, unfindableClientLibraries
+  };
+
+  private static final String[] outputStatements = {
+      "SUCCESS - The following %s libraries had the latest version of google-cloud-shared-dependencies: ",
+      "FAIL - The following %s libraries did not contain any version of google-cloud-shared-dependencies: ",
+      "FAIL - The following %s libraries had outdated versions of google-cloud-shared-dependencies: ",
+      "FAIL - The following %s libraries had unfindable POM files: "
+  };
+
+  public static void main(String[] args) throws ParseException, MavenRepositoryException {
+    String latestCommitMessage = getLatestCommitMessage();
+    if (latestCommitMessage == null) {
+      System.out.println("Commit message does not update dependencies! Returning success!");
+      System.exit(0);
+      return;
+    }
+
+    DefaultArtifact latestSharedDependencies = new DefaultArtifact("com.google.cloud",
+        "google-cloud-shared-dependencies", null, null);
+    ArtifactData sharedDependenciesData = ArtifactData
+        .generateArtifactData(latestSharedDependencies);
+    String latestSharedDependenciesVersion = sharedDependenciesData.getLatestVersion();
+    System.out.println("The latest version of google-cloud-shared-dependencies is "
+        + latestSharedDependenciesVersion);
+
+    if (latestCommitMessage.contains(updateDependency)) {
+      String dependencyStart = latestCommitMessage
+          .substring(latestCommitMessage.indexOf("com.google.cloud:"));
+      //Should be of the form ["groupId:artifactId", "to", "vX.X.X"]
+      String[] items = dependencyStart.split(" ");
+      //We already know the groupId
+      String groupId = "com.google.cloud";
+      String artifactId = items[0].split("")[1];
+      String version = items[2].substring(1);
+
+      Artifact dependencyArtifact = new DefaultArtifact(groupId, artifactId, null, version);
+      String sharedDependenciesVersion = ArtifactData.generateArtifactData(dependencyArtifact)
+          .getSharedDependenciesVersion();
+      if (sharedDependenciesVersion == null) {
+        System.out.println("Update " + dependencyStart + " failed due to an unfindable POM!");
+        System.exit(1);
+      } else if (sharedDependenciesVersion.isEmpty()) {
+        System.out.println("Update " + dependencyStart
+            + " failed due to lacking google-cloud-shared-dependencies");
+        System.exit(1);
+      } else if (sharedDependenciesVersion.equals(latestSharedDependenciesVersion)) {
+        System.out.println("Update " + dependencyStart
+            + " succeeded with google-cloud-shared dependencies version "
+            + sharedDependenciesVersion);
+        System.exit(0);
```

Comment:
possibly safer to return from main rather than calling System.exit

Reply:
The intention behind calling System.exit was to produce either a success or error code. The return from this block is slightly further down in the code